src/Arcus.WebApi.Correlation/IServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using GuardNet;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Arcus.WebApi.Correlation 
+{
+    /// <summary>
+    /// Adds operation and transaction correlation to the application.
+    /// </summary>
+    // ReSharper disable once InconsistentNaming
+    public static class IServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds the correlation related services to the dependency injection system of the application.
+        /// </summary>
+        public static IServiceCollection AddCorrelation(this IServiceCollection services, Action<CorrelationOptions> configureOptions)
```

Comment:
Configuring the correlation should be optional, no?
```suggestion
        public static IServiceCollection AddCorrelation(this IServiceCollection services, Action<CorrelationOptions> configureCorrelationOptions = null)
```

Reply:
Yes, that's why there's an overload for this.

Reply:
Good for me!